types/es-abstract/2016/SameValueNonNumber.d.ts

```diff
@@ -0,0 +1,11 @@
+/**
+ * @param x
+ * @param y
+ *
+ * @throws {TypeError} If `x` or `y` is a `number` or they're different types.
+ */
+declare function SameValueNonNumber(
+    x: object | null | undefined | string | boolean | symbol,
+    y: object | null | undefined | string | boolean | symbol,
```

Comment:
```suggestion
    y: typeof x,
```

Reply:
That won’t work, since `typeof x` is `string | boolean | symbol | object | null | undefined` and not the type of whatever was passed as the first parameter.